### ISSUE TITLE
RedundantRegexpQuantifiers is incorrect with interpolated values

### ISSUE THREAD

**Issue by jdelStrother** [rubocop/rubocop#12024]
The new Lint/RedundantRegexpQuantifiers rule (#11565) thinks that this regexp:

```ruby
FOO = /foo/
BAR = /a*#{FOO}?/x
```

ought to be corrected to:

```ruby
FOO = /foo/
BAR = /a*#{FOO}/x
```

which unless I'm missing something is not the same thing.

--------

## Expected behavior


No warnings for the above regexp


## Actual behavior

```
bundle exec rubocop --debug foo.rb                                                                                                                                                                                                                                                                                                   Mon  3 Jul 17:42:40 2023
For /Users/jon/Developer/web: configuration from /Users/jon/Developer/web/.rubocop.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-capybara-2.18.0/config/default.yml
Default configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-1.54.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-capybara-2.18.0/lib/../config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-factory_bot-2.22.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-factory_bot-2.22.0/lib/../config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-performance-1.18.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-performance-1.18.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rails-2.20.2/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rails-2.20.2/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rspec-2.22.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.2.0/gems/rubocop-rspec-2.22.0/config/default.yml
Inheriting configuration from /Users/jon/Developer/web/.rubocop_todo.yml
Inspecting 1 file
Scanning /Users/jon/Developer/web/foo.rb
Loading cache from /Users/jon/.cache/rubocop_cache/48171ced22e8e3296ed3e6f7fd0eceeed315852b/54104447a3375cf77f8576dbe02772b138da069d/5807de7a226f560f395da1b40085f52f19a101a7
W

Offenses:

foo.rb:4:9: W: [Correctable] Lint/RedundantRegexpQuantifiers: Replace redundant quantifiers * and ? with a single *.
BAR = /a*#{FOO}?/x
        ^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.324270000011893 seconds
```

## Steps to reproduce the problem

Create a file containing:

```ruby
# frozen_string_literal: true

FOO = /foo/
BAR = /a*#{FOO}?/x
```

and run rubocop on it.

## RuboCop version

```
$ bundle exec rubocop -V
1.54.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin21]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.22.0
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
```
